### ISSUE TITLE
azure-search-documents/samples/sample_authentication.py fails on aad

### ISSUE THREAD

**Issue by olsgaard** [Azure/azure-sdk-for-python#31827]
- **Package Name**: azure-search
- **Package Version**: 11.4.0b6 and 11.3.0
- **Operating System**: Ubuntu
- **Python Version**: Python 3.11.0

**Describe the bug**
When running https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/search/azure-search-documents/samples/sample_authentication.py, `authentication_with_aad()` fails, despite having logged in with `az login`

**To Reproduce**
login and then export, export variables
```
az login
export AZURE_SEARCH_SERVICE_ENDPOINT="https://my-search-service.search.windows.net/"  
export AZURE_SEARCH_INDEX_NAME=myindex
export AZURE_SEARCH_API_KEY=$(az search admin-key show --service-name=my-search-service -g rg-my-search --query 'primaryKey' -o tsv)
python sample_authentication.py 
```

returns following error:

```
There are 0 documents in the myindex search index.
Traceback (most recent call last):
  File "/home/sample_authentication.py", line 88, in <module>
    authentication_with_aad()
  File "/home/sample_authentication.py", line 68, in authentication_with_aad
    result = search_client.get_document_count()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/myenv/lib/python3.11/site-packages/azure/core/tracing/decorator.py", line 78, in wrapper_use_tracer
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/myenv/lib/python3.11/site-packages/azure/search/documents/_search_client.py", line 107, in get_document_count
    return int(self._client.documents.count(**kwargs))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/myenv/lib/python3.11/site-packages/azure/core/tracing/decorator.py", line 78, in wrapper_use_tracer
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/myenv/lib/python3.11/site-packages/azure/search/documents/_generated/operations/_documents_operations.py", line 489, in count
    raise HttpResponseError(response=response, model=error)
azure.core.exceptions.HttpResponseError: () Authorization failed.
Code: 
Message: Authorization failed.
```

**Expected behavior**
We expect the sample script to print `There are 0 documents in the myindex search index.` twice as it authenticates and retrieves document count twice.

**Screenshots**

**Additional context**
`az login` is obviously successful, as the call to `az search admin-key show` returns a valid key, as seen by the first printing of "There are 0 documents in the myindex search index."

I don't know if this is an error in the search sample file, or in `DefaultAzureCredential()` or in the search client.

**Comment by kashifkhan** [comment]
Thank you for the feedback @olsgaard . We will investigate and get back to you asap.

**Comment by xiangyan99** [comment]
1.0.0b2 is too old. I don't think we have added the AAD auth in that version.

Could you try latest versions of azure-search-documentation library?

Thanks.

**Comment by olsgaard** [comment]
Thank you for your quick reply @xiangyan99 

I checked my installed version by typing `pip show azure-search`, where I should have used `pip show azure-search-documents`

Confusingly these are 2 different packages. I have removed the `azure-search` package and re-tested.

My azure-search-documents is version 11.4.0b6 and I have also tested with 11.3.0, which is the newest release version and I get the same error.

I have updated the issue to reflect this.
